src/v/cluster/tm_stm.cc

```diff
@@ -19,29 +19,85 @@
 
 #include <seastar/core/coroutine.hh>
 #include <seastar/core/future.hh>
+#include <seastar/util/bool_class.hh>
 
 #include <filesystem>
 #include <optional>
 
 namespace cluster {
 
-static model::record_batch serialize_tx(tm_transaction tx) {
+namespace {
+
+using use_new_version_tx_bool = ss::bool_class<struct use_new_version_tx_tag>;
```

Comment:
replace 'new' with the actual version? if we move to a v2, v1 is not new anymore.